src/main/java/org/folio/search/controller/SearchConsortiumController.java

```diff
@@ -82,11 +97,72 @@ public ResponseEntity<ConsortiumLocationCollection> getConsortiumLocations(Strin
       .totalRecords(result.getTotalRecords()));
   }
 
-  private void checkAllowance(String tenantHeader) {
+  @Override
+  public ResponseEntity<ConsortiumHolding> getConsortiumHolding(UUID id, String tenantHeader) {
+    var tenant = verifyAndGetTenant(tenantHeader);
+    var holdingId = id.toString();
+    var searchRequest = idCqlRequest(tenant, INSTANCE_HOLDING_FIELD_NAME, holdingId);
+
+    var result = searchService.getConsortiumHolding(holdingId, searchRequest);
+    return ResponseEntity.ok(result);
+  }
+
+  @Override
+  public ResponseEntity<ConsortiumItem> getConsortiumItem(UUID id, String tenantHeader) {
+    var tenant = verifyAndGetTenant(tenantHeader);
+    var itemId = id.toString();
+    var searchRequest = idCqlRequest(tenant, INSTANCE_ITEM_FIELD_NAME, itemId);
+
+    var result = searchService.getConsortiumItem(itemId, searchRequest);
+    return ResponseEntity.ok(result);
+  }
+
+  @Override
+  public ResponseEntity<ConsortiumHoldingCollection> fetchConsortiumBatchHoldings(String tenantHeader,
+                                                                                  BatchIdsDto batchIdsDto) {
+    var tenant = verifyAndGetTenant(tenantHeader);
+    var holdingIds = batchIdsDto.getIds().stream().map(UUID::toString).collect(Collectors.toSet());
+    var searchRequest = idsCqlRequest(tenant, INSTANCE_HOLDING_FIELD_NAME, holdingIds);
+
+    var result = searchService.fetchConsortiumBatchHoldings(searchRequest, holdingIds);
+    return ResponseEntity.ok(result);
+  }
+
+  @Override
+  public ResponseEntity<ConsortiumItemCollection> fetchConsortiumBatchItems(String tenantHeader,
+                                                                            BatchIdsDto batchIdsDto) {
+    if (batchIdsDto.getIds().isEmpty()) {
+      return ResponseEntity
+        .ok(new ConsortiumItemCollection());
+    }
+
+    var tenant = verifyAndGetTenant(tenantHeader);
+    var itemIds = batchIdsDto.getIds().stream().map(UUID::toString).collect(Collectors.toSet());
+    var searchRequest = idsCqlRequest(tenant, INSTANCE_ITEM_FIELD_NAME, itemIds);
+
+    var result = searchService.fetchConsortiumBatchItems(searchRequest, itemIds);
+    return ResponseEntity.ok(result);
+  }
+
+  private String verifyAndGetTenant(String tenantHeader) {
     var centralTenant = consortiumTenantService.getCentralTenant(tenantHeader);
     if (centralTenant.isEmpty() || !centralTenant.get().equals(tenantHeader)) {
       throw new RequestValidationException(REQUEST_NOT_ALLOWED_MSG, XOkapiHeaders.TENANT, tenantHeader);
     }
+    return centralTenant.get();
+  }
```

Comment:
I don't like methods like doSomeAndDoSomeAnd...
May be better way to do something like next
```
private String getVerifiedTenant(String tenantHeader) {
    verifyTenant(tenantHeader)
    return consortiumTenantService.getCentralTenant(tenantHeader).get();
}

private void verifyTenant(String tenantHeader) {
    var centralTenant = consortiumTenantService.getCentralTenant(tenantHeader);
    if (centralTenant.isEmpty() || !centralTenant.get().equals(tenantHeader)) {
      throw new RequestValidationException(REQUEST_NOT_ALLOWED_MSG, XOkapiHeaders.TENANT, tenantHeader);
    }
}
```